### PR TITLE
whitesur-cursors: init at unstable-2022-06-17

### PR DESCRIPTION
**Pull request by sei40kr**
###### Description of changes

Added [whitesur-cursors](https://github.com/vinceliuice/WhiteSur-cursors).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/icons/whitesur-cursors/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ fetchFromGitHub, lib, stdenvNoCC }:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "WhiteSur-cursors";
```
Comment (SuperSandro2000) [inline pkgs/data/icons/whitesur-cursors/default.nix:4]:
```suggestion
  pname = "whitesur-cursors";
```

File: pkgs/data/icons/whitesur-cursors/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{ fetchFromGitHub, lib, stdenvNoCC }:
+
+stdenvNoCC.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/data/icons/whitesur-cursors/default.nix:3]: If we are just copying files we can also use only fetchFromGitHub and postFetch.

File: pkgs/data/icons/whitesur-cursors/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  installPhase = ''
+    mkdir -p $out/share/icons
+    cp -r dist $out/share/icons/WhiteSur-cursors
+  '';
+
+  meta = with lib; {
+    description = "WhiteSur cursors theme for linux desktops";
+    homepage = "https://github.com/vinceliuice/WhiteSur-cursors";
+    platforms = platforms.linux;
+    license = licenses.gpl3;
```
Comment (romildo) [inline pkgs/data/icons/whitesur-cursors/default.nix:23]: `gpl3Only` or `gpl3Plus`?

**Review by romildo — CHANGES_REQUESTED** [review]

File: pkgs/data/icons/whitesur-cursors/default.nix (hunk 2 above)
Comment (sei40kr) [reply pkgs/data/icons/whitesur-cursors/default.nix:3]: @SuperSandro2000 
I read `fetchFromGitHub` implementation and i noticed it's buggy.

When we specify `leaveDotGit = true`, it uses `fetchgit` as the fetcher and we can't specify `meta` together.
And when we don't specify `leaveDotGit`, it uses `fetchzip` as the fetcher and we can't specify `postFetch` together.

File: pkgs/data/icons/whitesur-cursors/default.nix (hunk 3 above)
Comment (sei40kr) [reply pkgs/data/icons/whitesur-cursors/default.nix]: @romildo 
`gpl3Only`. Thanks!

File: pkgs/data/icons/whitesur-cursors/default.nix (hunk 1 above)
Comment (sei40kr) [reply pkgs/data/icons/whitesur-cursors/default.nix]: @SuperSandro2000 
Thanks. I've fixed it.

**Review by romildo — APPROVED** [review]

File: pkgs/data/icons/whitesur-cursors/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/data/icons/whitesur-cursors/default.nix:3]: For fetchzip you need to use extraPostFetch https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/fetchzip/default.nix#L14 and fetchgit I am patching right now.

Comment (sei40kr) [reply pkgs/data/icons/whitesur-cursors/default.nix]: @SuperSandro2000 
Thanks!
I've updated the commit.

**Comment by gepbird** [comment]
Superseded by https://github.com/NixOS/nixpkgs/pull/249078

**Pull request closed without merging**